plugins/module_utils/pulp_helper.py

```diff
@@ -225,15 +228,16 @@ def chunked_upload(self, path, sha256, size):
                         end=offset + actual_chunk_size - 1,
                         size=size,
                     )
-                    with TemporaryDirectory() as temp_dir:
-                        chunk_file_name = os.path.join(temp_dir, 'chunk.bin')
-                        with open(chunk_file_name, 'wb') as chunk_file:
-                            chunk_file.write(chunk)
-                        upload = self.uploads_api.update(
-                            upload_href=upload.pulp_href,
-                            file=chunk_file_name,
-                            content_range=content_range,
-                        )
+                    temp_dir = mkdtemp(dir="/tmp")
+                    chunk_file_name = os.path.join(temp_dir, 'chunk.bin')
+                    with open(chunk_file_name, 'wb') as chunk_file:
+                        chunk_file.write(chunk)
+                    upload = self.uploads_api.update(
+                        upload_href=upload.pulp_href,
+                        file=chunk_file_name,
```

Comment:
I think we should wrap this in a try finally construct. Also it might be valuable to look, whether ansible provides something there.

Reply:
> Thank you for this first of all contributions. I haven't thought, python 2 was still that important...
> This is not a full review yet, but what I can say already is that we need to find a way to run those tests in py2 also. Do you want to have a go there?

I guess when it comes to a lot of OS releases their stable repositories still use python2(.7) a lot.
How about adding python2.7 to the build pipeline? (https://docs.travis-ci.com/user/languages/python/#specifying-python-versions). I went ahead and added a commit. Results seem fine: https://travis-ci.com/mdellweg/ansible_modules_pulp/builds/138780423
Locally, it will be built using the available `python` executable. I think you can change it by using e.g. `python2.7 -m pytest`.

Reply:
Do you mean the temporary directory? I'll add a commit.

Reply:
Oh yes. Sorry this mobile app seems to be bad at adding comments to the right lines.